lib/inspec/control_eval_context.rb

```diff
@@ -133,7 +134,16 @@ def to_s
 
         define_method :only_if do |&block|
           return unless block
-          return if @skip_file == true || block.yield == true
+          return if @skip_file == true
+
+          begin
+            return if block.yield == true
+          rescue => e
+            # Ignore error if a mock connection
+            # Example: `inspec check` with `only_if { os.name.include?('windows' }`
+            return if @backend.mock_transport?
+            raise e
```

Comment:
You can eliminate the `e` here - `raise` within a `rescue` will re-raise the original exception.